### PR TITLE
PIMS-1467: parcel hooks in useParcelApi

### PR DESCRIPTION
**Pull request by LawrenceLau2020**
## 🎯 Summary

  
[PIMS-1467: ](https://citz-imb.atlassian.net/browse/PIMS-1467) - Implenenting parcel hooks in useParcelApi

## 🔰 Checklist

- [x] I have read and agree with the following checklist and am following the guidelines in our [Code of Conduct](CODE_OF_CONDUCT.md) document.

> - I have performed a self-review of my code.
> - I have commented my code, particularly in hard-to-understand areas.
> - I have made corresponding changes to the documentation where required.
> - I have tested my changes to the best of my ability.
> - My changes generate no new warnings.

### CODE REVIEW

**Comment by GrahamS-Quartech** [comment]
I think a function for adding a parcel should also be part of this ticket, because we will need to call that when hitting submit on the add property form.

File: react-app/src/hooks/api/useParcelsApi.ts [hunk 1]
```diff
@@ -1,52 +1,45 @@
...
+  ProjectNumbers: string;
+  Corporation: string;
+  Ownership: 50;
+  IsSensitive: true;
+  UpdatedOn: Date;
+  Evaluations: Evaluation | null;
+  Fiscals: Fiscal | null;
+}
+
+const useParcelsApi = (absoluteFetch: IFetch) => {
```
Comment (GrahamS-Quartech) [inline react-app/src/hooks/api/useParcelsApi.ts:35]: These functions should probably have the above interface in their return signature.

**Comment by LawrenceLau2020** [comment]
I've made changes to the return signature, let me know if I've missed something, I wonder if we also need a patch for updating partial parcel fields, as well as needing a getParcelByPID?

File: react-app/src/hooks/api/useParcelsApi.ts [hunk 2]
```diff
@@ -1,56 +1,68 @@
...
+  const addParcel = async () => {
+    const { parsedBody } = await absoluteFetch.post('/parcels');
+    return parsedBody as Parcel;
+  };
+  const updateParcel = async () => {
+    const { parsedBody } = await absoluteFetch.put('/parcels');
+    return parsedBody as Parcel;
+  };
```
Comment (dbarkowsky) [inline react-app/src/hooks/api/useParcelsApi.ts:44]: Should this be ById? The controller is expecting that in the params.
We also need to pass a body.
```suggestion
  const updateParcel = async (id: number, parcel: Partial<Parcel>) => {
    const { parsedBody } = await absoluteFetch.put(`/parcels/${id}`, parcel);
    return parsedBody as Parcel;
  };
```

File: react-app/src/hooks/api/useParcelsApi.ts [hunk 3]
```diff
@@ -1,56 +1,68 @@
...
+  UpdatedOn: Date;
+  Evaluations: Evaluation[] | null;
+  Fiscals: Fiscal[] | null;
+}
+
+const useParcelsApi = (absoluteFetch: IFetch) => {
+  const addParcel = async () => {
+    const { parsedBody } = await absoluteFetch.post('/parcels');
+    return parsedBody as Parcel;
+  };
```
Comment (dbarkowsky) [inline react-app/src/hooks/api/useParcelsApi.ts:40]: Needs a body passed. This body is what we hope will be added in the API.
```suggestion
  const addParcel = async (parcel: Parcel) => {
    const { parsedBody } = await absoluteFetch.post('/parcels', parcel);
    return parsedBody as Parcel;
  };
```

File: react-app/src/hooks/api/useParcelsApi.ts [hunk 4]
```diff
@@ -1,56 +1,68 @@
...
+  Agency: Agency | null;
+  Address1: string;
+  ProjectNumbers: string;
+  Corporation: string;
+  Ownership: string;
+  IsSensitive: true;
+  UpdatedOn: Date;
+  Evaluations: Evaluation[] | null;
+  Fiscals: Fiscal[] | null;
+}
```
Comment (dbarkowsky) [inline react-app/src/hooks/api/useParcelsApi.ts:34]: Some of these might not be necessary, like PID and PIN. 
Maybe we should mark some as optional?
e.g.
``` 
PID?: number;
PIN?: number;
```

Comment (LawrenceLau2020) [reply react-app/src/hooks/api/useParcelsApi.ts:34]: ah, good catch.

File: react-app/src/hooks/api/useParcelsApi.ts [hunk 5]
```diff
@@ -29,16 +29,16 @@ export interface Parcel {
   Ownership: string;
   IsSensitive: true;
   UpdatedOn: Date;
-  Evaluations: Evaluation[] | null;
-  Fiscals: Fiscal[] | null;
+  Evaluations?: Evaluation[] | null;
+  Fiscals?: Fiscal[] | null;
 }
```
Comment (LawrenceLau2020) [inline react-app/src/hooks/api/useParcelsApi.ts:34]: also made Evaluations and Fiscals nullable

**Review by GrahamS-Quartech — APPROVED** [review]

**Pull request merged**